src/renderer/screens/Editor/Macros/MacroEditor.js

```diff
@@ -0,0 +1,156 @@
+// -*- mode: js-jsx -*-
+/* Chrysalis -- Kaleidoscope Command Center
+ * Copyright (C) 2020-2022  Keyboardio, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import Focus from "@api/focus";
+import { KeymapDB } from "@api/keymap";
+
+import Box from "@mui/material/Box";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import CardHeader from "@mui/material/CardHeader";
+import IconButton from "@mui/material/IconButton";
+import Stack from "@mui/material/Stack";
+
+import CloseIcon from "@mui/icons-material/Close";
+
+import MacroStep from "./MacroStep";
+import MacroStepAdd from "./MacroStepAdd";
+import MacroStepEditor from "./MacroStepEditor";
+
+const focus = new Focus();
+const db = new KeymapDB();
+
+const MacroEditor = (props) => {
+  const { macroId, macro, onMacroChange, macroStep, setMacroStep } = props;
+  const [stepEditorOpen, setStepEditorOpen] = useState(false);
+  const { t } = useTranslation();
+
+  const onStepDelete = (index) => {
+    const m = macro.map((v) => Object.assign({}, v));
```

Comment:
Where do we get lowercase `macro` from? I don't think we likely want to be throwing something with quite so generic a name into our global namespace without an explicit import. (And also, not being able to track it down in code will be..frustrating for folks working on the codebase.)

We might also want to think about renaming this thing to something like KaleidoscopeMacro or ..somethign?

Reply:
> Where do we get lowercase `macro` from? 

A few lines up:

`const { macroId, macro, onMacroChange, macroStep, setMacroStep } = props;`

From `props`.

> I don't think we likely want to be throwing something with quite so generic a name into our global namespace

We're not, it's coming from props. We could use a better name for either `m` or `macro`, however. `newMacro` instead of `m` sounds fine to me. The purpose of `m` is to clone the current macro, adjust it according to the edit, and then replace the original with it. We need to clone it, because otherwise we end up editing the original, and React's state diffing won't kick in.

Reply:
I just missed it in props. Thank you.